sqlglot/generator.py

```diff
@@ -1852,7 +1852,7 @@ def interval_sql(self, expression: exp.Interval) -> str:
 
         if self.SINGLE_STRING_INTERVAL:
             this = expression.this.name if expression.this else ""
-            return f"INTERVAL '{this}{unit}'"
+            return f"INTERVAL '{this}{unit}'" if this else f"INTERVAL{unit}"
```

Comment:
Needed because we can have `value::interval day`, which would be transpiled to `cast(value as interval ' day')`.